main.go

```diff
@@ -46,6 +49,32 @@ func check(e error) {
 	}
 }
 
+// DRY - move logic out of clone functions
+func executeclone(repo *github.Repository, directory string, wg *sync.WaitGroup) error {
+	urlToClone := ""
+	switch *protocol {
+	case "https":
+		urlToClone = *repo.CloneURL
+	case "ssh":
+		urlToClone = *repo.SSHURL
+	default:
+		fmt.Println("Unrecognized protocol: '" + *protocol + "'")
+		os.Exit(2)
+	}
+
+	// do not clone forks
+	if !*cloneForks && *repo.Fork {
+		return nil
+	} else {
+		// clone it
+		wg.Add(1)
+		fmt.Println(urlToClone)
+		// thread out the git clone
+		go gitclone(urlToClone, directory, wg)
```

Comment:
shouldn't this be `&wg` i.e. sending the waitgroup pointer just like others?
Also, shouldn't we be waiting as well by using `wg.Wait()`

Reply:
`wg` is already a pointer (`*wg` in the function arguments). Passing it without `&` here is passing a pointer.